jerry-core/api/jerry-snapshot.c

```diff
@@ -42,12 +42,12 @@ snapshot_get_global_flags (bool has_regex, /**< regex literal is present */
 
   uint32_t flags = 0;
 
-#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+#if defined (JERRY_BUILTIN_REGEXP) && (JERRY_BUILTIN_REGEXP == 1)
```

Comment:
As I see there is a sanity check macro guard for the incorrect configuration:

```c
#if !defined (JERRY_BUILTIN_REGEXP) \
|| ((JERRY_BUILTIN_REGEXP != 0) && (JERRY_BUILTIN_REGEXP != 1))
# error "Invalid value for JERRY_BUILTIN_REGEXP macro."
#endif
```
Therefore wouldn't be much easier to use only the

```c
#if JERRY_BUILTIN_REGEXP == 1
//code
#endif
```

guard all over the codebase? I think these double checks could make the code much harder to read especially if two or more guards are combined [e.g.](https://github.com/pando-project/jerryscript/pull/2793/files#diff-8ffb0036b085bcd4039b9adcb34f9ad2R1049).